### PR TITLE
show parent comment in comment review panel

### PR DESCRIPTION
**Pull request by penk85**
WPC-275: Show the parent of comment being reviewed in the comment review panel
parentcomment

### CODE REVIEW

File: packages/editor/src/client/routes/commentList.tsx [hunk 1]
```diff
@@ -146,6 +149,19 @@ export function CommentList() {
...
+
+  const fetchParentComment = useCommentQuery({
+    variables: {id: currentComment?.parentID}
+  }).data?.comment?.revisions[length - 1]
+
+  const [currentCommentParent, setCurrentCommentParent] = useState<CommentRevision | undefined>(
+    fetchParentComment
+  )
+
+  useEffect(() => {
```
Comment (bishoymelek) [inline packages/editor/src/client/routes/commentList.tsx:161]: Have you tried fetching together with the comments list? what do you think?

**Review by bishoymelek — CHANGES_REQUESTED** [review]
Great job :tada: just one point

File: packages/editor/src/client/routes/commentList.tsx (hunk 1 above)
Comment (roosnic1) [reply packages/editor/src/client/routes/commentList.tsx:161]: @bishoymelek how would that work? We don't know which parent comment to fetch without the user clicking on a comment.

Comment (bishoymelek) [reply packages/editor/src/client/routes/commentList.tsx:161]: If we added resolve function to parentID, it should work fine.

Comment (roosnic1) [reply packages/editor/src/client/routes/commentList.tsx:161]: You might be right. @penk85 and myself already discussed this. I think there are some circular reference issues. So for now this is the easiest solution.

Comment (penk85) [reply packages/editor/src/client/routes/commentList.tsx:161]: @bishoymelek there were issues with resolving a comment inside a comment, and I haven't found a way around that.

File: packages/editor/src/client/routes/commentList.tsx [hunk 2]
```diff
@@ -348,23 +364,54 @@ export function CommentList() {
...
+                              parentComment?.comment?.revisions[length - 1].createdAt
+                            ).toLocaleString()}
+                          </div>
                           <RichTextBlock
-                            disabled
                             displayOnly
+                            displayOneLine
+                            disabled
                             // TODO: remove this
                             onChange={console.log}
```
Comment (roosnic1) [inline packages/editor/src/client/routes/commentList.tsx:384]: you can remove the TODO comment and replace console.log with `/* do nothing */`

File: packages/editor/src/client/routes/commentList.tsx [hunk 3]
```diff
@@ -348,23 +364,54 @@ export function CommentList() {
...
+                  <Timeline align="left">
+                    {currentComment?.revisions?.length
+                      ? currentComment?.revisions?.map(({text, createdAt}, i) => (
+                          <Timeline.Item key={i}>
+                            <div>{new Date(createdAt).toLocaleString()}</div>
+                            <RichTextBlock
+                              disabled
+                              displayOnly
+                              // TODO: remove this
+                              onChange={console.log}
```
Comment (roosnic1) [inline packages/editor/src/client/routes/commentList.tsx:407]: same here

**Review by roosnic1 — CHANGES_REQUESTED** [review]
looks awesome just some tiny changes.

File: packages/editor/src/client/routes/commentList.tsx (hunk 1 above)
Comment (bishoymelek) [reply packages/editor/src/client/routes/commentList.tsx:161]: @penk85 

file:packages/api/src/graphql/comment.ts
line:118
`export const GraphQLComment: GraphQLObjectType<Comment, Context> = new GraphQLObjectType<
  Comment,
  Context
>({
  name: 'Comment',
  fields: () => ({
    id: {type: GraphQLNonNull(GraphQLID)},

    user: {
      type: GraphQLNonNull(GraphQLUser),
      resolve: createProxyingResolver(({userID}, _, {dbAdapter}) => {
        return dbAdapter.user.getUserByID(userID)
      })
    },
    authorType: {type: GraphQLNonNull(GraphQLCommentAuthorType)},

    itemID: {type: GraphQLNonNull(GraphQLID)},
    itemType: {
      type: GraphQLNonNull(GraphQLCommentItemType)
    },

    parentComment: {
      type: GraphQLComment,
      resolve: createProxyingResolver(({parentID}, _, {dbAdapter}) => {
        if (parentID) return dbAdapter.comment.getCommentById(parentID)
        return null
      })
    },

    revisions: {type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLCommentRevision)))},

    state: {type: GraphQLNonNull(GraphQLCommentState)},

    rejectionReason: {type: GraphQLCommentRejectionReason},

    createdAt: {type: GraphQLNonNull(GraphQLDateTime)},
    modifiedAt: {type: GraphQLNonNull(GraphQLDateTime)}
  })
})
`

File: packages/editor/src/client/routes/commentList.tsx (hunk 2 above)
Comment (penk85) [reply packages/editor/src/client/routes/commentList.tsx:384]: @roosnic1, ts won't let me assign an empty value here.

File: packages/api/src/graphql/query.private.ts [hunk 4]
```diff
@@ -431,6 +437,18 @@ export const GraphQLQuery = new GraphQLObjectType<undefined, Context>({
...
     // =======
+
+    comment: {
+      type: GraphQLComment,
+      args: {id: {type: GraphQLID}},
+      resolve(root, {id}, {authenticate, dbAdapter}) {
+        const {roles} = authenticate()
+        authorise(CanGetComment, roles)
+
+        return dbAdapter.comment.getCommentById(id)
```
Comment (bishoymelek) [inline packages/api/src/graphql/query.private.ts:448]: I think we should check if id does not equal null?

example from same file:

`
 if (id == null) {
   throw new UserInputError('You must provide `id`')
}
`

File: packages/editor/src/client/routes/commentList.tsx [hunk 5]
```diff
@@ -146,6 +148,20 @@ export function CommentList() {
 
   const [isConfirmationDialogOpen, setConfirmationDialogOpen] = useState(false)
   const [currentComment, setCurrentComment] = useState<Comment>()
+
+  const commentParentVariables = {
```
Comment (bishoymelek) [inline packages/editor/src/client/routes/commentList.tsx:152]: can we name as parentCommentVariables for consistency?

**Review by bishoymelek — COMMENTED** [review]
Amazing work @penk85 , I added 2 simple points

File: packages/editor/src/client/routes/commentList.tsx (hunk 5 above)
Comment (bishoymelek) [reply packages/editor/src/client/routes/commentList.tsx:152]: @penk85 any updates on this one? :eyes:

**Review by bishoymelek — DISMISSED** [review]
:tada: Great work @penk85

**Review by HanaAyman — DISMISSED** [review]

**Review by roosnic1 — CHANGES_REQUESTED** [review]
Looks awesome. just one little indentation fix and you're ready to merge.

**Review by roosnic1 — APPROVED** [review]
Awesome. 🎉

**Review by bishoymelek — APPROVED** [review]
:tada: great one @penk85

**Review by HanaAyman — APPROVED** [review]

**Pull request merged**